### PR TITLE
MGMT-1358: validate cluster name on register and update

### PR DESCRIPTION
**Pull request by danielerez**
Added validation for cluster name format in register/update cluster flows.

### CODE REVIEW

**Comment by danielerez** [comment]
@filanov @romfreiman

File: internal/cluster/validations/validations.go [hunk 1]
```diff
@@ -153,3 +157,17 @@ func checkDNSRecordsExistence(names []string, dnsProvider dnsproviders.Provider)
...
 }
+
+// ValidateClusterName validates specified name format and cluster's existance in DB
+func ValidateClusterName(name string, db *gorm.DB) error {
+	if matched, err := regexp.MatchString(clusterNameRegex, name); !matched {
+		return fmt.Errorf("Cluster name is not valid: %v", err)
+	}
+
+	var cluster common.Cluster
+	err := db.First(&cluster, "name = ?", name).Error
```
Comment (avishayt) [inline internal/cluster/validations/validations.go:168]: Why do we need cluster name to be unique?  I assume this is for DNS, but if so, I would validate it there.  Why do we care if two clusters are named "test" if they use different DNS?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -310,6 +310,10 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 		}
 		setPullSecret(&cluster, params.NewClusterParams.PullSecret)
 	}
+	if err := validations.ValidateClusterName(swag.StringValue(params.NewClusterParams.Name), b.db); err != nil {
+		return installer.NewRegisterClusterBadRequest().
```
Comment (filanov) [inline internal/bminventory/inventory.go:314]: You can use `common.NewApiError(http.StatusBadRequest, errors.New("Cluster name already exists"))`

File: internal/bminventory/inventory_test.go [hunk 3]
```diff
@@ -783,6 +783,17 @@ var _ = Describe("cluster", func() {
 			Expect(reply).To(BeAssignableToTypeOf(installer.NewUpdateClusterNotFound()))
 		})
 
+		It("Invalid cluster name", func() {
```
Comment (filanov) [inline internal/bminventory/inventory_test.go:786]: Usually we mock non local packages, so i think that this test belong to subsystem.

File: internal/cluster/validations/validation_test.go [hunk 4]
```diff
@@ -132,6 +134,39 @@ var _ = Describe("Base DNS validation", func() {
...
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+	It("invalid name", func() {
+		err := ValidateClusterName("test!A", db)
+		Expect(err).Should(HaveOccurred())
+	})
+	It("existing name", func() {
+		err := ValidateClusterName("test", db)
+		Expect(err).Should(HaveOccurred())
+	})
```
Comment (filanov) [inline internal/cluster/validations/validation_test.go:160]: Probably should add after each and close the DB.

File: internal/cluster/validations/validations.go (hunk 1 above)
Comment (filanov) [reply internal/cluster/validations/validations.go:168]: +1

Comment (danielerez) [reply internal/cluster/validations/validations.go:168]: > Why do we need cluster name to be unique? I assume this is for DNS, but if so, I would validate it there. Why do we care if two clusters are named "test" if they use different DNS?

Cluster name uniqueness is actually already validated in the UI, the validation was just missing in the backend. But it's also needed for DNS indeed, in the cluster name is part of the prefix. So don't we want to correlate with the UI validation and validate the cluster name?

Comment (danielerez) [reply internal/cluster/validations/validations.go]: removed uniqueness check for now, left the format name validation.

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (danielerez) [reply internal/bminventory/inventory.go]: done

File: internal/bminventory/inventory_test.go (hunk 3 above)
Comment (danielerez) [reply internal/bminventory/inventory_test.go]: removed mock for now.

File: internal/cluster/validations/validation_test.go (hunk 4 above)
Comment (danielerez) [reply internal/cluster/validations/validation_test.go:155]: removed for now.

**Pull request merged**